libcodechecker/server/api/report_server.py

```diff
@@ -2048,10 +2036,14 @@ def massStoreRun(self, name, tag, version, b64zip, force):
 
                 LOG.debug("Storing done for report " + str(report_id))
 
-            if len(report_path_and_events):
-                store_handler.changePathAndEvents(self.__storage_session,
-                                                  run_id,
-                                                  report_path_and_events)
+        for bug_hash, reports in hash_map_reports.items():
+            if bug_hash in new_bug_hashes:
+                session.query(Report) \
```

Comment:
Maybe it would be better to collect new report ids to a list and call this delete operation only once.